### ISSUE TITLE
ChangeHandler - postpone animation until view and data are ready

### ISSUE THREAD

**Issue by MFlisar** [bluelinelabs/Conductor#118]
Can someone shortly explain on which basis the `ChangeHandler` animations are started? I assume they start as soon as the view is ready.

I have some working code without conductor, that transitions from a `RecyclerView` to another `RecyclerView` (they share 1 to 4 images). This means, the animation must wait for following things:

* second view must be finished
* data for second view must be ready
* 1 - 4 images must have finished loading

Is there any way to achieve this sort of animation with `conductor`? Can I somehow manually start the transition as soon as the data is ready instead of having the framework decide?

**Comment by sockeqwe** [comment]
Sure, you can do that manually. This is what ChangeHandlers are for.

Take a look at some of the default implementation of change handlers:

https://github.com/bluelinelabs/Conductor/tree/develop/conductor/src/main/java/com/bluelinelabs/conductor/changehandler

There is a class ControllerChangeHandler they extend from.

https://github.com/bluelinelabs/Conductor/blob/develop/conductor/src/main/java/com/bluelinelabs/conductor/ControllerChangeHandler.java

They have to override a method called performChange() which as last
parameter takes a ControllerChangeCompletedListener as parameter.

So do your stuff and once your new controller is ready to run the animation
you controller has to notify your custom changehandler to call
ControllerChangeCompletedListener.onChangeCompleted().

MFlisar <notifications@github.com> schrieb am Di., 13. Sep. 2016, 13:46:

**Comment by MFlisar** [comment]
Thanks. That's working so far.

One further question: we must set push and pop change handler seperately, but the handler can distinct between a push and a pop. Therefore I think it's save to use the same handler for push and pop, correct?

Like following:

    ControllerChangeHandler changeHandler = ...;
    mController.getRouter().pushController(RouterTransaction.with(...)
                    .pushChangeHandler(changeHandler )
                    .popChangeHandler(changeHandler));
